Java/src/org/imos/ddb/JDBCDDB.java

```diff
@@ -33,160 +33,193 @@
 import java.sql.Connection;
 import java.sql.DriverManager;
 import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
 import java.sql.SQLException;
 import java.sql.Statement;
 import java.util.ArrayList;
 import java.util.List;
 
 /**
- * In memory representation of the IMOS deployment database using JDBC. 
- * This class uses a generic JDBC driver and connection string so is
- * database agnostic, allowing the deployment database to be hosted on any
- * JDBC supported database platform.
+ * In memory representation of the IMOS deployment database using JDBC. This
+ * class uses a generic JDBC driver and connection string so is database
+ * agnostic, allowing the deployment database to be hosted on any JDBC supported
+ * database platform.
  * 
  * This class should never be instantiated directly; use the static method
  * org.imos.ddb.DDB.getDDB.
  * 
  * @author Gordon Keith <gordon.keith@csiro.au>
+ * @author Peter Jansen <peter.jansen@csiro.au> - generic database schema changes 
  * 
  * @see http://java.sun.com/javase/6/docs/technotes/guides/jdbc/bridge.html
  */
-public class JDBCDDB extends DDB {
-  
-  /**The JDBC database driver*/
-  private String driver;
-  private String connection;
-  private String user;
-  private String password;
-  
-  /**
-   * Create a JDBC DDB object using the specified JDBC driver and database connection. 
-   * 
-   * @param driver Class name of JDBC database driver
-   * @param connection Database connection string, must include user and password if required by the database.
-   * @throws ClassNotFoundException If the specified Database driver can't be found
-   * @throws SQLException If an attempt to open a connection to the database fails
-   */
-  protected JDBCDDB(String driver, String connection, String user, String password) throws ClassNotFoundException, SQLException {
-	  this.driver = driver;
-	  this.connection = connection;
-	  this.user = user;
-	  this.password = password;
-	  
-	  // Test connection - throws exception at creation if can't connect
-      Class.forName(driver);
-      
-      Connection conn = DriverManager.getConnection(connection, user, password);
-      conn.close();
-  }
+public class JDBCDDB extends DDB

```

Comment:
The code was so ungly to read I could not figure out what was going on.Once this PR is merged the code will be much more readable. 
Any technical issues with this way of dealing with the database tables?
Does the code work for you on your version of the database?